lib/provisioning/ProvisionDispatcher.js

```diff
@@ -0,0 +1,356 @@
+                                 error: err });
+                return;
+            }
+            this.isLeader = this._amILeader(children);
+            this.log.debug('updated leader',
+                           { zkPath: this.zkEndpoint + zkPath,
+                             children,
+                             iamLeader: this.isLeader });
+            if (this.isLeader) {
+                this._redispatchProvisions();
+            }
+        });
+    }
+
+    _onOwnerChange() {
+        const zkPath = ProvisionDispatcher.constants.OWNERS;
+        this.client.getChildren(zkPath, event => {
+            this.log.debug('got owner event',
+                           { zkPath: this.zkEndpoint + zkPath,
+                             event });
+            this._onOwnerChange();
+        }, (err, children) => {
+            if (err) {
+                this.log.error('list failed',
+                               { zkPath: this.zkEndpoint + zkPath,
+                                 error: err });
+                return;
+            }
+            this.owners = children;
+            this.log.debug('owners updated',
+                           { zkPath: this.zkEndpoint + zkPath,
+                             owners: this.owners });
+            if (this.isLeader) {
+                this._redispatchProvisions();
+            }
+        });
+    }
+
+    _onMyselfChange(cb) {
+        // monitor change in my content
+        const myPath = `${ProvisionDispatcher.constants.OWNERS}/` +
+                  `${this.myName}`;
+        this.client.getData(
+            myPath, event => {
+                this.log.debug('got owner self event',
+                               { zkPath: this.zkEndpoint + myPath,
+                                 event });
+                this._onMyselfChange(cb);
+            },
+            (err, data) => {
+                if (err &&
+                    err.getCode() !== zookeeper.Exception.NO_NODE) {
+                    this.log.error('error in getData',
+                                   { zkPath: this.zkEndpoint + myPath,
+                                     error: err });
+                    return undefined;
+                }
+                if (data !== undefined) {
+                    const provisionList = JSON.parse(data);
+                    this.log.info('provisioning update',
+                                  { zkPath: this.zkEndpoint + myPath,
+                                    provisionList });
+                    return cb(null, provisionList);
+                }
+                return undefined;
+            }
+        );
+    }
+
+    _onProvisionChange() {
+        const zkPath = ProvisionDispatcher.constants.PROVISIONS;
+        this.client.getChildren(zkPath, event => {
+            this.log.debug('got provision event', {
+                zkPath: this.zkEndpoint + zkPath,
+                event,
+            });
+            this._onProvisionChange();
```

Comment:
This is only triggered when there's an actual change on the children (the second argument of `getChildren` is an event watcher, not the actual callback).